packages/segmented-control/src/SegmentedControlOption.tsx

```diff
@@ -79,6 +80,15 @@ export const SegmentedControlOption = forwardRef<
       didComponentMount.current = true;
     }, [focused, followFocus, usingKeyboard, isfocusInComponent]);
 
+    /**
+     * Wraps text in a `<span>`. This span will be used to clip text and add an ellipsis. We do this so that consumers don't have to add the `<span>` themselves. We can't add the ellipsis styles to the parent container because that container has `display: flex` which doesn't work with `text-overflow: ellipsis`.
+     */
+    const renderedChildren = React.Children.map(children, child => {
```

Comment:
@TheSonOfThomp What do you think about this? if its text then we wrap it.

Reply:
Yeah this makes sense! I'm on the fence whether it makes more sense to assign this, or just put this block inside the JSX, but this looks good 👍 
```jsx
<span className={labelStyle}>
  {React.Children.map(children, child => {
    if (typeof child === 'string')
      return <span className={textEllipsisStyle}>{child}</span>;
    return child;
  })}
</span>
```

Reply:
it might be easier to read inline, i'll move it